src/sources/tcp.rs

```diff
@@ -40,36 +36,9 @@ impl TcpConfig {
     }
 }
 
-inventory::submit! {
-    SourceDescription::new_without_default::<TcpConfig>("tcp")
-}
-
-#[typetag::serde(name = "tcp")]
-impl SourceConfig for TcpConfig {
-    fn build(
-        &self,
-        _name: &str,
-        _globals: &GlobalOptions,
-        out: mpsc::Sender<Event>,
-    ) -> crate::Result<super::Source> {
-        let tcp = RawTcpSource {
-            config: self.clone(),
-        };
-        tcp.run(self.address, self.shutdown_timeout_secs, out)
-    }
-
-    fn output_type(&self) -> DataType {
-        DataType::Log
-    }
-
-    fn source_type(&self) -> &'static str {
-        "tcp"
-    }
-}
-
 #[derive(Debug, Clone)]
-struct RawTcpSource {
-    config: TcpConfig,
+pub struct RawTcpSource {
```

Comment:
I don't love having to make this public as it seems kind of like an implementation detail, but the socket source needs access to it to establish a concrete TCP socket, so the options seemed either to make it public here or move its definition into socket.rs, but it seemed nice to keep all the TCP functionality together here in tcp.rs so I went with just making it public.

Reply:
If we end up merging this into `socket.rs` or a submodule thereof, we should be able to avoid this.

Reply:
So making tcp a submodule of socket didn't let me get rid of this change, but now that the entire tcp module is private that doesn't matter as much anymore?